printer_zpl2/models/printing_label_zpl2_component.py

```diff
@@ -0,0 +1,144 @@
+from odoo import fields, models
+
+_logger = logging.getLogger(__name__)
+
+try:
+    import zpl2
+except ImportError:
+    _logger.debug('Cannot `import zpl2`.')
+
+
+class PrintingLabelZpl2Component(models.Model):
+    _name = 'printing.label.zpl2.component'
+    _description = 'ZPL II Label Component'
+    _order = 'sequence'
+
+    label_id = fields.Many2one(
+        comodel_name='printing.label.zpl2', string='Label',
+        required=True, ondelete='cascade', help='Label using this component.')
+    sequence = fields.Integer(help='Order used to print the elements.')
+    name = fields.Char(required=True, help='Name of the component.')
+    origin_x = fields.Integer(
+        required=True, default=10,
+        help='Origin point of the component in the label, X coordinate.')
+    origin_y = fields.Integer(
+        required=True, default=10,
+        help='Origin point of the component in the label, Y coordinate.')
+    component_type = fields.Selection(
+        selection=[
+            ('text', 'Text'),
+            ('rectangle', 'Rectangle / Line'),
+            ('circle', 'Circle'),
+            (zpl2.BARCODE_CODE_11, 'Code 11'),
+            (zpl2.BARCODE_INTERLEAVED_2_OF_5, 'Interleaved 2 of 5'),
+            (zpl2.BARCODE_CODE_39, 'Code 39'),
+            (zpl2.BARCODE_CODE_49, 'Code 49'),
+            (zpl2.BARCODE_PDF417, 'PDF417'),
+            (zpl2.BARCODE_EAN_8, 'EAN-8'),
+            (zpl2.BARCODE_UPC_E, 'UPC-E'),
+            (zpl2.BARCODE_CODE_128, 'Code 128'),
+            (zpl2.BARCODE_EAN_13, 'EAN-13'),
+            ('sublabel', 'Sublabel'),
+        ], string='Type', required=True, default='text', oldname='type',
+        help='Type of content, simple text or barcode.')
+    font = fields.Selection(
+        selection=[
+            (zpl2.FONT_DEFAULT, 'Default'),
+            (zpl2.FONT_9X5, '9x5'),
+            (zpl2.FONT_11X7, '11x7'),
+            (zpl2.FONT_18X10, '18x10'),
+            (zpl2.FONT_28X15, '28x15'),
+            (zpl2.FONT_26X13, '26x13'),
+            (zpl2.FONT_60X40, '60x40'),
+            (zpl2.FONT_21X13, '21x13'),
+        ], required=True, default=zpl2.FONT_DEFAULT,
+        help='Font to use, for text only.')
+    thickness = fields.Integer(help='Thickness of the line to draw.')
+    color = fields.Selection(
+        selection=[
+            (zpl2.COLOR_BLACK, 'Black'),
+            (zpl2.COLOR_WHITE, 'White'),
+        ], default=zpl2.COLOR_BLACK,
+        help='Color of the line to draw.')
+    orientation = fields.Selection(
+        selection=[
+            (zpl2.ORIENTATION_NORMAL, 'Normal'),
+            (zpl2.ORIENTATION_ROTATED, 'Rotated'),
+            (zpl2.ORIENTATION_INVERTED, 'Inverted'),
+            (zpl2.ORIENTATION_BOTTOM_UP, 'Read from Bottom up'),
+        ], required=True, default=zpl2.ORIENTATION_NORMAL,
+        help='Orientation of the barcode.')
+    check_digits = fields.Boolean(
+        help='Check if you want to compute and print the check digit.')
```

Comment:
It's called check digit in the documentation : https://www.zebra.com/content/dam/zebra/manuals/en-us/software/zpl-zbi2-pm-en.pdf (page 1299).